include/drivers/flash.h

```diff
@@ -165,12 +165,32 @@ static inline int z_impl_flash_erase(struct device *dev, off_t offset,
  *  @brief  Enable or disable write protection for a flash memory
  *
  *  This API is required to be called before the invocation of write or erase
- *  API. Please note that on some flash components, the write protection is
+ *  API. Behaviour at calls to @ref flash_write @ref flash_erase without
```

Comment:
Just a language tweak. Rather than saying "driver-dependent", I would use the term "undefined behavior".

```
Any calls to flash_write() or flash_erase() that do not first disable write protection using this function result in undefined behavior.
```

Undefined behavior is of course different that implementation-defined behavior (which is what @pabigot advocated). I think that UB is the best choice here to allow driver writers to continue to adjust things without breaking well-behaved programs.

Reply:
done